models/DataObject/Concrete/Dao.php

```diff
@@ -278,12 +278,8 @@ public function update(bool $isUpdate = null): void
                 }
             }
         }
-
-        if ($isUpdate) {
-            Helper::insertOrUpdate($this->db, 'object_store_' . $this->model->getClassId(), $data);
-        } else {
-            $this->db->insert('object_store_' . $this->model->getClassId(), Helper::quoteDataIdentifiers($this->db, $data));
-        }
+        $tableName = 'object_store_' . $this->model->getClassId();
+        Helper::upsert($this->db, $tableName, $data, $this->getPrimaryKey($tableName));
```

Comment:
Isn't that a change of behavior? 
Previously, there was an exception thrown, if `$isUpdate === false` and ID was already in DB. Now, it silently updates the existing record.

Reply:
True, if `$isUpdate` is set to false and the key is already existing in the table it wouldn´t throw an exception anymore. But then the question would be how that can happen. 

Anyway, I reverted the change and substituted the `Helper::insertOrUpdate` call only.